src/ls.js

```diff
@@ -46,119 +48,51 @@ function _ls(options, paths) {
 
   if (!paths)
     paths = ['.'];
-  else if (typeof paths === 'object')
-    paths = paths; // assume array
   else if (typeof paths === 'string')
     paths = [].slice.call(arguments, 1);
 
-  var list = [];
+  paths = common.expand(paths, { dot: options.all });
```

Comment:
Why not perform this in common.wrap? This doesn't give correct behavior if you do it here like this. Doing it in `common.wrap` is the easiest quick fix.

```Bash
$ ls -A
.secret.txt foo.txt bar.txt
$ ls *.txt # this only shows non-hidden files
foo.txt bar.txt
$ cat test.js
require('shelljs/global');
echo(ls('*.txt')); // this shouldn't output secret files either
$ node test.js # but it actually does
.secret.txt foo.txt bar.txt
```